guard/src/commands/parse_tree.rs

```diff
@@ -21,51 +21,50 @@ impl Command for ParseTree {
         PARSE_TREE
     }
 
-    fn command(&self) -> App<'static> {
-        App::new(PARSE_TREE)
-            .about(
-                r#"Prints out the parse tree for the rules defined in the file.
-"#,
-            )
+    fn command(&self) -> clap::Command {
+        clap::Command::new(PARSE_TREE)
+            .about("Prints out the parse tree for the rules defined in the file.")
             .arg(
-                Arg::with_name(RULES.0)
+                Arg::new(RULES.0)
                     .long(RULES.0)
                     .short(RULES.1)
-                    .takes_value(true)
-                    .help("Provid a rules file")
+                    .help("Provide a rules file")
+                    .action(ArgAction::Set)
                     .required(false),
             )
             .arg(
-                Arg::with_name(OUTPUT.0)
+                Arg::new(OUTPUT.0)
                     .long(OUTPUT.0)
                     .short(OUTPUT.1)
-                    .takes_value(true)
                     .help("Write to output file")
+                    .action(ArgAction::Set)
                     .required(false),
             )
             .arg(
-                Arg::with_name(PRINT_JSON.0)
+                Arg::new(PRINT_JSON.0)
                     .long(PRINT_JSON.0)
                     .short(PRINT_JSON.1)
-                    .required(false)
+                    .action(ArgAction::SetTrue)
```

Comment:
Do we want really want to remove `.required(false)` here? That will make this flag mandatory no?

Reply:
since its a flag, i.e. its default value is false we dont need it to be required.

from the docs

```Pro tip: Flags (i.e. not positional, or arguments that take values) shouldn’t be required by default. This is because if a flag were to be required, it should simply be implied. No additional information is required from user. Flags by their very nature are simply boolean on/off switches. The only time a user should be required to use a flag is if the operation is destructive in nature, and the user is essentially proving to you, “Yes, I know what I’m doing.”```